db/compaction.cc

```diff
@@ -405,20 +405,23 @@ void Compaction::Summary(char* output, int len) {
 uint64_t Compaction::OutputFilePreallocationSize() const {
   uint64_t preallocation_size = 0;
 
+  for (const auto& level_files : inputs_) {
+    for (const auto& file : level_files.files) {
+      preallocation_size += file->fd.GetFileSize();
+    }
+  }
+
   if (max_output_file_size_ != port::kMaxUint64 &&
-      (cfd_->ioptions()->compaction_style == kCompactionStyleLevel ||
+      (immutable_cf_options_.compaction_style == kCompactionStyleLevel ||
        output_level() > 0)) {
-    preallocation_size = max_output_file_size_;
-  } else {
-    for (const auto& level_files : inputs_) {
-      for (const auto& file : level_files.files) {
-        preallocation_size += file->fd.GetFileSize();
-      }
-    }
+    preallocation_size = std::min(max_output_file_size_, preallocation_size);
   }
+
   // Over-estimate slightly so we don't end up just barely crossing
   // the threshold
-  return preallocation_size + (preallocation_size / 10);
+  // No point to prellocate more than 1GB.
```

Comment:
I'd like to understand more about this? 
Is it because files greater than 1GB are rare or do we have a limit somewhere that says RocksDB cannot have files more than 1 GB?

Reply:
The reason that we preallocate in SST file is to avoid extends too small. 1GB is already a significant size for an extend, so I don't see a value of preallocate larger than that.